src/internal.c

```diff
@@ -7785,6 +7785,15 @@ int InitSSL(WOLFSSL* ssl, WOLFSSL_CTX* ctx, int writeDup)
             return MEMORY_E;
         }
         XMEMSET(ssl->param, 0, sizeof(WOLFSSL_X509_VERIFY_PARAM));
+
+        /* pass on PARAM flags value from ctx to ssl */
+        if (wolfSSL_X509_VERIFY_PARAM_set_flags(wolfSSL_get0_param(ssl),
+            (unsigned long)wolfSSL_X509_VERIFY_PARAM_get_flags(
+            wolfSSL_CTX_get0_param(ctx))) != WOLFSSL_SUCCESS) {
+
+            XFREE(ssl->param, ssl->heap, DYNAMIC_TYPE_OPENSSL);
+            return WOLFSSL_FAILURE;
+        }
```

Comment:
Don't `XFREE`. Just log and let `FreeSSL` take care of `free`'ing anything that got allocated.